pandas/tests/frame/test_alter_axes.py

```diff
@@ -229,13 +226,24 @@ def test_set_index_verify_integrity(self, frame_of_index_cols):
     def test_set_index_raise(self, frame_of_index_cols, drop, append):
         df = frame_of_index_cols
 
-        with tm.assert_raises_regex(KeyError, '.*'):  # column names are A-E
+        with tm.assert_raises_regex(KeyError, "['foo', 'bar', 'baz']"):
+            # column names are A-E
             df.set_index(['foo', 'bar', 'baz'], drop=drop, append=append)
 
         # non-existent key in list with arrays
-        with tm.assert_raises_regex(KeyError, '.*'):
+        with tm.assert_raises_regex(KeyError, 'X'):
             df.set_index([df['A'], df['B'], 'X'], drop=drop, append=append)
 
+        rgx = 'The parameter "keys" may only contain a combination of.*'
```

Comment:
rgx -> msg

Reply:
@jreback Changed. Please let me know what you want me to do about the allowed argument signature. Allowing all list-likes also means checking `ndim==1` for `np.ndarrays`, etc.

See also:
https://github.com/pandas-dev/pandas/pull/22486#discussion_r219705345
https://github.com/pandas-dev/pandas/pull/22486#discussion_r219705993